core/global/display_manager.gd

```diff
@@ -40,97 +31,275 @@ func supports_brightness() -> bool:
 
 ## Sets the brightness on all discovered backlights to the given value as a 
 ## percentage (e.g. 1.0 is 100% brightness)
-func set_brightness(value: float, type: VALUE = VALUE.ABSOLUTE) -> int:
+func set_brightness(value: float, type: VALUE = VALUE.ABSOLUTE, backlight: String = "") -> int:
 	if not supports_brightness():
 		logger.debug("Failed to set brightness, no providers found")
 		return FAILED
-	value = minf(value, 1.0)
-	value = maxf(value, 0.0)
-	backlights = get_backlight_paths()
-	
-	# Set the brightness for all backlights
-	for backlight_path in backlights:
-		# Calculate the value based on whether this is an absolute or relative change
-		var abs_value := value
-		if type == VALUE.RELATIVE:
-			abs_value = get_brightness(backlight_path) + value
-			abs_value = minf(abs_value, 1.0)
-			abs_value = maxf(abs_value, 0.0)
-		var real_value := int(abs_value * get_max_brightness_value(backlight_path))
-		
-		# Use the preferred method of setting brightness
-		var brightness_file := "/".join([backlight_path, "brightness"])
-		logger.debug("Setting brightness on " + brightness_file + " to " + str(real_value))
-		if brightness_provider == BRIGHTNESS_PROVIDER.STEAMOS:
-			if _steamos_priv_write(brightness_file, real_value) != OK:
-				return FAILED
-			continue
-		
-		return FAILED
-	
+
+	var err := brightness_provider.set_brightness(value, type, backlight)
+	if err != OK:
+		return err
 	brightness_changed.emit()
-	
-	return OK
+
+	return err
 
 
 ## Returns the current brightness level for the given backlight as a percent
 func get_brightness(backlight_path: String) -> float:
-	var cur_brightness := get_brightness_value(backlight_path)
-	var max_brightness := get_max_brightness_value(backlight_path)
-	if cur_brightness == -1 or max_brightness == -1:
-		return -1
-	return float(cur_brightness) / max_brightness
+	if brightness_provider == null:
+		return -1.0
```

Comment:
Why is this return the only float?

Reply:
Because `get_brightness_value()` and `get_max_brightness_value()` return the raw brightness values, whereas this function returns the normalized brightness value between `0.0` and `1.0`.